src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1047,7 +1062,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                 // and will not be reported to callers Compilation.GetDiagnostics()
 
                 ImmutableArray<SourceSpan> dynamicAnalysisSpans = ImmutableArray<SourceSpan>.Empty;
-                bool hasBody = flowAnalyzedBody != null;
+                bool hasBody = flowAnalyzedBody != null && methodSymbol.HasBody;
```

Comment:
Do we have a test where `methodSymbol.HasBody` returns `true` here, but `flowAnalyzedBody` is not null?